bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -360,31 +363,40 @@ void checkLayout () {
 * 	Break paragraphs into lines, wraps the text, and initialize caches.
 */
 void computeRuns (GC gc) {
+	int newNativeZoom = getNativeZoom(gc);
+	if (nativeZoom != newNativeZoom) {
+		nativeZoom = newNativeZoom;
+		freeRuns();
+	}
 	if (runs != null) return;
 	long hDC = gc != null ? gc.handle : device.internal_new_GC(null);
 	long srcHdc = OS.CreateCompatibleDC(hDC);
 	allRuns = itemize();
 	for (int i=0; i<allRuns.length - 1; i++) {
 		StyleItem run = allRuns[i];
-		OS.SelectObject(srcHdc, getItemFont(run));
-		shape(srcHdc, run);
+		OS.SelectObject(srcHdc, getItemFont(run, gc));
+		shape(gc, srcHdc, run);
 	}
 	SCRIPT_LOGATTR logAttr = new SCRIPT_LOGATTR();
 	SCRIPT_PROPERTIES properties = new SCRIPT_PROPERTIES();
-	int lineWidth = indent, lineStart = 0, lineCount = 1;
+	int wrapIndentInPixels = DPIUtil.autoScaleUp(wrapIndent, getZoom(gc));
```

Comment:
Yes we would do that after we have merged everything atleast all the multi zoom level PRs so that its easy to refactor the calls. Right now they are spread across multiple PRs. And we just need to refactor them in the end.

Reply:
Perfect, I agree.